index.src.html

```diff
@@ -87,14 +98,17 @@
     to schedule work.
 
 <dfn lt="running a control message|control message steps">Running a control
-    message</dfn> means executing a sequence of steps specified by the method
+    message</dfn> means performing a sequence of steps specified by the method
     that enqueued the message.
 
+The steps of a control message may depend on <dfn>injected state</dfn>, supplied
+by the method that enqueud the message.
+
 <dfn lt="Run the control message processing loop">Running the control message
-    processing loop</dfn> means executing these steps.
+    processing loop</dfn> means performing these steps.
 1. While the control message queue is not <a>empty</a>
     1. Let |front message| be the next <a>dequeued</a> <a>control message</a>
-    2. If |front message| cannot be executed now, return.
+    2. If |front message| cannot be performd now, return.
```

Comment:
> when ? This is run manually from the main or worker thread, so there need to be a task queued on an event loop to do it, by something. Or maybe the intent is for authors to manually call, say, decode or flush to keep things rolling?

Its intended to be an implementation detail. Authors cannot drive it. Example: say author has called decode() 100 times, but the underlying hardware codec only supports 3 parallel decodes. We can process the first 3, but then we need to wait for the internal pipeline to finish one before dequeueing the next decode() request. 

There are many other cases where the UA's implementation may be internally unable to progress until some pending internal operation completes. In chromium's decoder code, you can see most of these [listed here](https://source.chromium.org/search?q=f:decoder_template.cc%20(%22Try%20again%22%20OR%20%22Processing%20continues%22)&ss=chromium%2Fchromium%2Fsrc).

> but you might also need to execute it to get a failure, and then to decide that this is a transient failure and that this should be retried later.

It is true that async execution of some control message may trigger a failure at a delay from when that control message is enqueued. This is what motivated the design to emit all failures via common error callback and to make all such failures fatal. The only recoverable errors are those where the user has obviously supplied invalid input (e.g. a decoder config with codedWidth = 0) - these throw exceptions immediately without altering the state of the codec. For fatal errors, authors should recover by recreating the codec and trying again. In the case of live/realtime, they would likely retry with a more recent input.

> What should author do when the processing loop is blocked? Stop attempting to submit Web Codec work? Continue, because it's a work queue? After all its purpose is to sequence work.

The only blockage the author should worry about is the scenario where their codec is falling behind real time. This occurs in the same scenarios we have w/ <audio> / <video> (e.g. 4k on a netbook without hardware acceleration). This is visible to the author in that `decodeQueueSize` starts to grow toward infinity. Latency sensitive apps should monitor its growth and adapt (e.g. lower resolution) if needed. 

The API is otherwise designed to be "fire and forget". Ex: you don't need to worry about whether we've completed settled the async aspects of your last `configure()` before calling `decode()`. Fire away, just watch your decodeQueueSize (if you care). 

> nit: performed.

Fixed.